src/bt/Node.cpp

```diff
@@ -114,4 +114,17 @@ std::string Node::status_print(Node::Status s) {
     }
 }
 
+void Node::setRole(std::string roleName) {
+    auto children = this->getChildren();
+    if (properties) {
+        this->properties->setString("ROLE", roleName);
+    }
+
+    for (auto child : this->getChildren()) {
+        if (child) {
```

Comment:
Are children sometimes null?... how

Reply:
They're shared pointers, their whole point is to always be valid until you dont use it anymore.